dashboard/src/actions/repos.ts

```diff
@@ -101,6 +101,30 @@ export const resyncRepo = (
   };
 };
 
+export const resyncAllRepos = (): ThunkAction<Promise<void>, IStoreState, null, AppReposAction> => {
+  return async (dispatch, getState) => {
+    try {
+      const {
+        config: { namespace },
+      } = getState();
+      // TODO: Do something to show progress
+      const repos = await AppRepository.list(namespace);
+      if (repos.items.length > 0) {
+        repos.items.forEach(repo => {
```

Comment:
here it's better to dispatch calls to the other method: `dispatch(resyncRepo(repo.metadata.name))` to avoid repeating code.

Reply:
+1, dispatching it will also make each resync request happen in parallel

Reply:
resyncRepo(name) function will make a GET and UPDATE API call, then a call to `AppRepository.list(namespace)`, so using it here will make many duplicate `AppRepository.list` calls.  Any more idea?
```
      const repo = await AppRepository.get(name, namespace);
      repo.spec.resyncRequests = repo.spec.resyncRequests || 0;
      repo.spec.resyncRequests++;
      await AppRepository.update(name, namespace, repo);
      // TODO: Do something to show progress
      dispatch(requestRepos());
      const repos = await AppRepository.list(namespace);
      dispatch(receiveRepos(repos.items));
```

Reply:
To be honest, I don't know why we are listing the repositories in the `resyncRepo` method, it's not doing anything. You can remove the lines 95-97 so we don't list them several times.

Reply:
I guess it is to reload the updated repos list?

Reply:
Hmm, good catch @jessehu, we definitely don't want to be listing multiple times. I think what we could do here is remove the last few lines from resyncRepo and reuse fetchRepos when calling this action. We can update RepoListContainer to do this:

https://github.com/kubeapps/kubeapps/blob/cc186fc257eb972326752b6f1b9960e5142e0330/dashboard/src/containers/RepoListContainer/RepoListContainer.ts#L28-L30

```
    resyncRepo: async (name: string) => {
      await dispatch(actions.repos.resyncRepo(name));
      return dispatch(actions.repos.fetchRepos());
    },
```

and then do the same for resyncAllRepos. What do you think?

Reply:
@prydonius With the new code above, the logic will be `resyncAllRepos` -> `dispatch(resyncRepo(name));`  -> `dispatch(actions.repos.fetchRepos());`, so seems it stills calls fetchRepos() multiple times ?  For now, is fetchRepos() needed after a resyncRepo(name)?  We only show repo name and URL on the UI, so I think it won't make any changes on UI unless there is a new app repo added/deleted in the backend by another kubeapps web UI client, but this is not expected goal of refreshing a single repo.